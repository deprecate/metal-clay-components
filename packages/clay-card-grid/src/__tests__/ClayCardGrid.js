import ClayCardGrid from '../ClayCardGrid';

let component;

let items = [
	{
		items: [
			{
				downloadHref: '#sabers',
				title: 'Sabers',
				type: 'folder',
			},
			{
				downloadHref: '#scripts',
				title: 'Scripts',
				type: 'folder',
			},
		],
		label: 'Folders',
	},
	{
		items: [
			{
				director: 'George Lucas',
				downloadHref: '#',
				image: './swepisode1.jpg',
				rating: 10,
				releaseDate: 'May 19th 1999',
				status: ['Error', 'Pending'],
				title: 'Episode I: The Phantom Menace',
				type: 'MOV',
			},
			{
				director: 'George Lucas',
				downloadHref: '#',
				rating: 50,
				releaseDate: 'May 12th 2002',
				status: ['Watched'],
				title: 'Episode II: Attack of the Clones',
				type: 'MOV',
			},
			{
				director: 'George Lucas',
				downloadHref: '#',
				image: './swepisode3.jpg',
				rating: 60,
				releaseDate: 'May 12th 2005',
				status: ['Watched'],
				title: 'Episode III: Revenge of the Sith',
				type: 'MOV',
			},
			{
				director: 'George Lucas',
				downloadHref: '#',
				image: './swepisode4.jpg',
				rating: 90,
				releaseDate: 'May 25th 1977',
				status: ['Watched'],
				title: 'Episode IV: A New Hope',
				type: 'MOV',
			},
			{
				director: 'Irvin Kershner',
				downloadHref: '#',
				image: './swepisode5.jpg',
				rating: 100,
				releaseDate: 'May 21th 1980',
				status: ['Watched'],
				title: 'Episode V: The Empire Strikes Back',
				type: 'MOV',
			},
			{
				director: 'Richard Marquand',
				downloadHref: '#',
				image: './swepisode6.jpg',
				rating: 90,
				releaseDate: 'May 25th 1983',
				status: ['Watched'],
				title: 'Episode VI: Return of the Jedi',
				type: 'MOV',
			},
			{
				director: 'J. J. Abrams',
				downloadHref: '#',
				image: './swepisode7.jpg',
				rating: 70,
				releaseDate: 'December 14th 2015',
				status: ['Watched'],
				title: 'Episode VII: The Force Awakens',
				type: 'MOV',
			},
			{
				director: 'Rian Johnson',
				downloadHref: '#',
				image: './swepisode8.jpg',
				rating: 0,
				releaseDate: 'December 9th 2017',
				status: ['Pending'],
				title: 'Episode VIII: The Last Jedi',
				type: 'MOV',
			},
		],
		label: 'Saga',
	},
	{
		items: [
			{
				director: 'Gareth Edwards',
				downloadHref: '#',
				image: './swrogueone.jpg',
				rating: 70,
				releaseDate: 'December 10th 2016',
				status: ['Watched'],
				title: 'Rogue One: A Star Wars Story',
				type: 'MKV',
			},
			{
				director: 'Ron Howard',
				downloadHref: '#',
				image: './swsolo.jpg',
				rating: 0,
				releaseDate: 'May 25th 2018',
				status: ['Pending'],
				title: 'Solo: A Star Wars Story',
				type: 'MKV',
			},
		],
		label: 'Expanded Universe',
	},
	{
		items: [
			{
				character: 'Darth Vader',
				initials: 'DV',
				side: 'dark',
				type: 'character',
			},
			{
				character: 'Emperor Palpatine',
				initials: 'EP',
				side: 'dark',
				type: 'character',
			},
			{
				character: 'Han Solo',
				initials: 'HS',
				side: 'light',
				type: 'character',
			},
			{
				character: 'Kylo Ren',
				initials: 'KR',
				side: 'dark',
				type: 'character',
			},
			{
				character: 'Leia Organa',
				initials: 'LO',
				side: 'light',
				type: 'character',
			},
			{
				character: 'Luke Skywalker',
				initials: 'LS',
				side: 'light',
				type: 'character',
			},
			{
				character: 'Yoda',
				initials: 'Y',
				side: 'light',
				type: 'character',
			},
		],
		label: 'Characters',
	},
];

const schema = [
	{
		contentRendererMap: {
			character: 'user',
			folder: 'horizontal',
			MKV: 'file',
		},
		fieldsMap: {
			contentRenderer: 'type',
			description: 'director',
			fileType: 'type',
			href: 'downloadHref',
			icon: 'type',
			imageAlt: 'title',
			imageSrc: 'image',
			initials: 'initials',
			labels: 'status',
			title: 'title',
			userColor: 'side',
			username: 'character',
		},
		fileTypeStylesMap: {
			MOV: 'danger',
		},
		iconsMap: {
			folder: 'folder',
			MOV: 'video',
			MKV: 'video',
		},
		labelStylesMap: {
			'Watched': 'success',
			'Pending': 'warning',
			'*': 'info',
		},
		userColorsMap: {
			light: 'primary',
			dark: 'danger',
		},
	},
];

const spritemap = '../node_modules/lexicon-ux/build/images/icons/icons.svg';

describe('ClayCardGrid', function() {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render the default markup', () => {
		component = new ClayCardGrid({
			items: items,
			schema: schema,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardGrid with classes', () => {
		component = new ClayCardGrid({
			elementClasses: 'my-custom-class',
			schema: [],
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardGrid with id', () => {
		component = new ClayCardGrid({
			id: 'myId',
			schema: [],
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a selectable items ClayCardGrid and emit an event on item toggle selection', () => {
		component = new ClayCardGrid({
			items: items,
			schema: schema,
			selectable: true,
			spritemap: spritemap,
		});

		const spy = jest.spyOn(component, 'emit');

		component.element.querySelector('ul li input[type="checkbox"]').click();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledWith('itemToggled', expect.any(Object));
	});
});
