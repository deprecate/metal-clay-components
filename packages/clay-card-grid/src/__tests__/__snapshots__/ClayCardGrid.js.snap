// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayCardGrid should render a ClayCardGrid with all kind items 1`] = `
<div>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">CL</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 1">User Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">MC</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 2">User Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">CM</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 3">User Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with all kind items and title 1`] = `
<div>
  <h4 class="card-section-header">All Items</h4>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 1">File Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 2">File Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 3">File Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 1">Folder 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 2">Folder 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 3">Folder 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 1" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 1">Image Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 2" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 2">Image Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 3" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 3">Image Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">CL</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 1">User Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">MC</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 2">User Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">CM</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 3">User Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with classes 1`] = `
<div class="my-custom-class">
  <ul class="card-page"></ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with file items 1`] = `
<div>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 1">File Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 2">File Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 3">File Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with file items and title 1`] = `
<div>
  <h4 class="card-section-header">File Items</h4>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 1">File Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 2">File Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card file-card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <title>documents-and-media</title>
              <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="File Item 3">File Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with folder items 1`] = `
<div>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 1">Folder 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 2">Folder 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 3">Folder 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with folder items and title 1`] = `
<div>
  <h4 class="card-section-header">Folder Items</h4>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 1">Folder 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 2">Folder 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-directory card card-horizontal">
        <div class="card-body">
          <div class="card-row">
            <span class="sticker sticker-unstyled">
              <span class="sticker-overlay">
                <svg aria-hidden="true" class="lexicon-icon lexicon-icon-folder">
                  <title>folder</title>
                  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#folder"></use>
                </svg>
              </span>
            </span>
            <div class="flex-col flex-col-expand flex-col-gutters">
              <div class="card-title text-truncate" title="Folder 3">Folder 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with id 1`] = `
<div id="myId">
  <ul class="card-page"></ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with image items 1`] = `
<div>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 1" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 1">Image Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 2" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 2">Image Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 3" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 3">Image Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with image items and title 1`] = `
<div>
  <h4 class="card-section-header">Image Items</h4>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 1" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 1">Image Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 2" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 2">Image Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card image-card">
        <div class="aspect-ratio card-item-first">
          <img alt="Image Item 3" class="aspect-ratio-item-center-middle aspect-ratio-item-fluid" src="./image1.jpg">
          <span class="sticker sticker-primary sticker-bottom-left">
            <span class="sticker-overlay">JPG</span>
          </span>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="Image Item 3">Image Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with user items 1`] = `
<div>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">CL</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 1">User Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">MC</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 2">User Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">CM</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 3">User Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render a ClayCardGrid with user items and title 1`] = `
<div>
  <h4 class="card-section-header">User Items</h4>
  <ul class="card-page">
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">CL</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 1">User Item 1</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">MC</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 2">User Item 2</div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li class="card-page-item-directory">
      <div class="card-type-asset card">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
            <span class="sticker sticker-primary rounded-circle sticker-xl">
              <span class="sticker-overlay">CM</span>
            </span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand">
              <div class="card-title text-truncate" title="User Item 3">User Item 3</div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ClayCardGrid should render the default markup 1`] = `
<div>
  <ul class="card-page"></ul>
</div>
`;
